lib/pure/os.nim

```diff
@@ -1688,13 +1688,28 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo =
     rawToFormalFileInfo(rawInfo, path, result)
 
 proc isHidden*(path: string): bool =
-  ## Determines whether a given path is hidden or not. Returns false if the
-  ## file doesn't exist. The given path must be accessible from the current
-  ## working directory of the program.
+  ## Determines whether the given ``path`` is hidden or not, using this reference https://en.wikipedia.org/wiki/Hidden_file_and_hidden_directory
   ##
-  ## On Windows, a file is hidden if the file's 'hidden' attribute is set.
-  ## On Unix-like systems, a file is hidden if it starts with a '.' (period)
-  ## and is not *just* '.' or '..' ' ."
+  ## On Windows: a path is hidden if it exists and it's "hidden" attribute is set.
+  ##
+  ## On posix: a hidden path starts with ``.`` (see examples).
+  ##
+  ## On OSX, in addition to the "dotfile" behaviour, existing paths with the
+  ## "Invisible" attribute are hidden in Finder, although not in ``ls``.
```

Comment:
Why are you discussing Finder's behaviour here? This is confusing because it implies that this function follows the Finder behaviour but I don't think that's the case.

Reply:
removed that comment (which was just to say there's a difference bw Finder's hidden vs ls's hidden, even though this function deals with ls's hidden)